### DISCUSSION TITLE
Unknown CMake Command "get_solution"

### DISCUSSION THREAD

**Discussion by dolgae03** [ANLAB-KAIST/KENSv3#96] [Project Q&A 4. Etc] [answered]
I followed all process in the document. but, i met this problem, and i can't find why it happened..

[proc] Executing command: /usr/bin/gcc -v
[proc] The command: ninja --version failed with error: Error: spawn ninja ENOENT
[proc] The command: ninja-build --version failed with error: Error: spawn ninja-build ENOENT
[main] Configuring folder: dolgae03 
[proc] Executing command: /usr/bin/cmake --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/usr/bin/gcc -DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/g++ -S/home/dolgae03/KENSv3-master/app/echo -B/home/dolgae03/build -G "Unix Makefiles"
[cmake] Not searching for unused variables given on the command line.
[cmake] -- Configuring incomplete, errors occurred!
[cmake] See also "/home/dolgae03/build/CMakeFiles/CMakeOutput.log".
[cmake] CMake Error at CMakeLists.txt:8 (get_solution):
[cmake]   Unknown CMake command "get_solution".
[cmake] 
[cmake] 
[cmake] CMake Warning (dev) in CMakeLists.txt:
[cmake]   No cmake_minimum_required command is present.  A line of code such as
[cmake] 
[cmake]     cmake_minimum_required(VERSION 3.16)
[cmake] 
[cmake]   should be added at the top of the file.  The version specified may be lower
[cmake]   if you wish to support older CMake versions for this project.  For more
[cmake]   information run "cmake --help-policy CMP0000".
[cmake] This warning is for project developers.  Use -Wno-dev to suppress it.
[cmake] 
[proc] The command: /usr/bin/cmake --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/usr/bin/gcc -DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/g++ -S/home/dolgae03/KENSv3-master/app/echo -B/home/dolgae03/build -G "Unix Makefiles" exited with code: 1 and signal: null

**Comment by whkim0** [comment]
1. Which version of CMake are you using? You need to use CMake 3.11 or higher for KENS.
Run `cmake --version` to check the version.
2. Please check whether you have accidently modified CMakeLists.txt.

**Reply by dolgae03** [reply]
the version of cmake is 3.16.3. 

i checked that it is the same as original file

Thank you for answering

**Reply by rubidam** [reply]
I solve this problem to install "build-essential"

try to write command `sudo apt-get install build-essential -y`

and try again.

**Reply by dolgae03** [reply]
I tried. but, it still doesn't work. :(

**Comment by antaechan** [comment]
Thanks! I solved it

**Comment by dolgae03** [comment, marked as answer]
I solved it. When you use Window + WSL, You need to set the root CMakeList.txt. If not, the error occured
